[a11y docn] Update switch component's readme for accessibility project (#4768)
BrightspaceUI/core: components/switch/README.md

| Property | Type | Description |
|---|---|---|

> | `text` | String, required | Acts as a  [primary label](https://www.w3.org/WAI/tutorials/forms/labels/) on the switch. Visible unless text-position is `hidden`. |

Reviewer: ```suggestion
| `text` | String, required | Acts as the [primary label](https://www.w3.org/WAI/tutorials/forms/labels/) for the switch. Visible unless text-position is `hidden`. |
```

  </ul>
</d2l-switch-visibility>
```

## Accessbility

> - When `text-position` is set to `hidden`, the required `text` will be used for the switch's hidden label so that it is [accessible](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA14) to non-sighted users.

Reviewer: Did you mean to use (from your comment) https://www.w3.org/WAI/WCAG22/quickref/#name-role-value instead? I think referencing the WCAG criteria itself is more valuable since it highlights the importance of the value, rather than the specific technique we've chosen, which could change any time.

Author: Yep, I copied the link from the wrong tab 😅 I'll replace it with that one 😄

> - When `text-position` is set to `hidden`, the switch will still make use of the `text` property as it is used to set the [`aria-label`](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6) of the component, so that screen readers can still pick up what the text of the component is.

Reviewer: Perhaps we should avoid referring to the internal implementation (i.e. Aria usage), since it's always subject to change.

```suggestion
- When `text-position` is set to `hidden`, the required `text` will be used for the switch's hidden label so that it is accessible to non-sighted users.
```

Author: So I was thinking of either adding a short sentence after this one that will include a link to https://www.w3.org/WAI/WCAG22/quickref/#name-role-value, mentioning that this is what it's helping us meet or alternatively add that link to the word `accessible` in this sentence instead. Do you think it's too extra and should just leave things as-is?

  </ul>
</d2l-switch-visibility>
```

## Accessbility

- When `text-position` is set to `hidden`, the required `text` will be used for the switch's hidden label so that it is [accessible](https://www.w3.org/WAI/WCAG22/quickref/#name-role-value) to non-sighted users.

> - `d2l-switch-visibility` is a notable exception to the best practice of having a static term as the label, this was done to provide more clarity in situations where there are external conditions that could influence whether or not the content in question is visible.

Reviewer 2: nit:
```suggestion
- `d2l-switch-visibility` is a notable exception to the best practice of having a static term as the label; this was done to provide more clarity in situations where there are external conditions that could influence whether or not the content in question is visible.
```
Or divide into two separate sentences.
